src/Extensions/Abstractions/Activities/TimedScope.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Omex.Extensions.Abstractions.Activities
+{
+	/// <summary>
+	/// Logs duration of activity
+	/// </summary>
+	/// <remarks>This class will be replaced with Activity after move to net 5, since it will be trackable and disposable</remarks>
+	public class TimedScope : IDisposable
+	{
+		/// <summary>
+		/// Creates TimedScope instance
+		/// </summary>
+		/// <param name="activity">activity connected to this timedscope</param>
+		/// <param name="result">TimedScope initial result</param>
+		public TimedScope(Activity activity, TimedScopeResult result)
+		{
+			Activity = activity;
+			this.SetResult(result);
+		}
+
+
+		/// <summary>
+		/// Starts TimedScope activity
+		/// </summary>
+		public TimedScope Start()
+		{
+			s_listener.StartActivity(Activity, this);
+			return this;
+		}
+
+
+		/// <summary>
+		/// Stop TimedScope and log informations about it
+		/// </summary>
+		public void Stop() => s_listener.StopActivity(Activity, this);
+
+
+		internal Activity Activity { get; }
```

Comment:
>internal [](start = 2, length = 8)

`internal` is a less usual access modifier. Would `protected` work?

Reply:
unfortunately not, it required in extensions methods, another option would be to make it public